src/frontend/apps/markdown/MarkdownViewer/index.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect } from 'react';
+
+import { Box, Heading } from 'grommet';
+import { defineMessages, useIntl } from 'react-intl';
+
+import { Nullable } from 'utils/types';
+import { Loader } from 'components/Loader';
+
+import { MarkdownAppData } from 'apps/markdown/data/MarkdownAppData';
+import LanguageSelector from '../LanguageSelector';
+import { MarkdownDocumentTranslation } from '../types/models';
+
+const messages = defineMessages({
+  // Messages
+  translationNotFound: {
+    defaultMessage: 'Translation not found',
+    description: 'Error message displayed when translation is not found',
+    id: 'component.MarkdownEditor.translationNotFound',
+  },
+});
+
+const MarkdownViewer = () => {
+  const intl = useIntl();
+
+  const markdownDocument = MarkdownAppData.markdownDocument!;
+
+  const [language, setLanguage] = React.useState('en');
+  const [htmlContent, setHtmlContent] = React.useState<Nullable<string>>(null);
+  const [documentTitle, setDocumentTitle] =
+    React.useState<Nullable<string>>(null);
+
+  const getTranslatedContent = (content: keyof MarkdownDocumentTranslation) => {
+    const translation = markdownDocument.translations.find(
+      (value) => value.language_code === language,
+    );
+    return translation
+      ? translation[content]
+      : intl.formatMessage(messages.translationNotFound);
+  };
+
+  useEffect(() => {
```

Comment:
You should move this hook usage with all other hook usage at the top of this component

Reply:
I'm not sure I can do that, the `useEffect` uses `getTranslatedContent` which is the only thing defined above, as your previous comment state it, I should use dependency on `getTranslatedContent` (instead of `language`) therefore it needs to be defined before, isn't it ?

Reply:
Yes indeed you have to define first the function `getTranslatedContent` :+1:  and use it as a dependency